src/main/java/net/ccbluex/liquidbounce/injection/forge/mixins/entity/MixinEntityLivingBase.java

```diff
@@ -170,4 +172,14 @@ private float hookBodyRotationsB(EntityLivingBase instance) {
 
         return instance instanceof EntityPlayerSP && Rotations.INSTANCE.shouldUseRealisticMode() && rotation != null ? rotation.getYaw() : instance.rotationYaw;
     }
+
+    /**
+     * @author SuperSkidder
+     * @reason Animations swing speed
+     */
+    @Overwrite
+    public int getArmSwingAnimationEnd(){
+        int speed = ModuleManager.INSTANCE.getModule(Animations.class).getState() ? (int) (2 + (20 - Animations.INSTANCE.getSwingSpeed())) : 6;
+        return this.isPotionActive(Potion.digSpeed) ? speed - (1 + this.getActivePotionEffect(Potion.digSpeed).getAmplifier()) : (this.isPotionActive(Potion.digSlowdown) ? speed + (1 + this.getActivePotionEffect(Potion.digSlowdown).getAmplifier()) * 2 : speed);
```

Comment:
What if the player has "Haste" and "Mining Fatigue" potion effects at the same time? =)

Reply:
Looks like it would ignore Mining Fatigue effect here. However that is how original Minecraft handle it. If you like maybe we can calculate the accurate speed of mining

Reply:
Well, if that's how original Minecraft handles it, I have no suggestions for that.